### PR TITLE
Added extra args to the istiod

### PR DESCRIPTION
**Pull request by fsadykov**
Hello Istio Comunity,

I was deploying the `istio` using helm by following the official documentation and I was not able to find the proper way of adding the extra arguments to `istiod` helm chart. It's very useful to add arguments see the example below

Deployment
```
      - args:
        - discovery
        - --monitoringAddr=:15014
        - --log_output_level=default:info
        - --domain
        - cluster.local
        - --keepaliveMaxServerConnectionAge
        - 30m
        - --tlsCertFile=/var/run/secrets/istiod/tls/tls.crt
        - --tlsKeyFile=/var/run/secrets/istiod/tls/tls.key
        - --caCertFile=/var/run/secrets/istiod/tls/ca.crt
```

The values.yaml
```
  extraArgs:
  - --tlsCertFile=/var/run/secrets/istiod/tls/tls.crt
  - --tlsKeyFile=/var/run/secrets/istiod/tls/tls.key
  - --caCertFile=/var/run/secrets/istiod/tls/ca.crt
```
By using this feature during the helm deployment I am adding my root ca and CA for the istiod to use it

Links
https://artifacthub.io/packages/helm/istio-official/istiod

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
/ok-to-test

**Comment by costinm** [comment]
I would suggest converting them to env variables or even better to mesh
config. We still have a couple of CLI args for Istiod - but not sure it's
worth the confusion.

**Comment by fsadykov** [comment]
@costinm It would be very nice if we could add the arguments. We are currently not using the `istioctl` or `istio-operator` to deploy it. So having the arguments definitely would help us to deploy properly. Regarding environment variables I was not able to find on official documentation 

https://istio.io/latest/docs/reference/commands/pilot-discovery/



We are using these arguments
```
  extraArgs:
  - --tlsCertFile=/var/run/secrets/istiod/tls/tls.crt
  - --tlsKeyFile=/var/run/secrets/istiod/tls/tls.key
  - --caCertFile=/var/run/secrets/istiod/tls/ca.crt
```

File: releasenotes/notes/istiod-helm-args.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: feature
+area: extensibility
+releaseNotes:
+- |
+  **Added** added extra args to istio controller helm chart.
```
Comment (ericvn) [inline releasenotes/notes/istiod-helm-args.yaml:6]: You don't need the double added.
Capitalize Istio.
Not sure if this is better under installation or extensibility.

**Comment by costinm** [comment]
Helm has a mechanism to update the env vars ( pilot.env ) - and we have a pretty large number of settings using env variables. We have a couple of remaining settings via args, but we never added a mechanism to specify them - in part to encourage devs to add any setting using either MeshConfig or env. 

I think with the exception of istioctl - none of the servers should require CLI args - which also means helm doesn't need 2 mechanism to customize the same kind of settings. 

So my suggestion was: if you need to configure one of the remaining options using CLI, it would be better to change the code to support env. My understanding was that all CLI args can also be set as env ( or someone has worked on this), so maybe that's already done, but that was long ago, don't remember the status.

**Comment by costinm** [comment]
@therealmitchconnors - do you remember this ? Viper or something ? 

We also discussed having all settings in MeshConfig or ProxyConfig - with documentation on which can be dynamically changed without restart ( both env variables and CLI require restart - but if we have the same settings in MeshConfig, some can be made dynamic ).

**Comment by therealmitchconnors** [comment]
Unfortunately, all the work to use viper to make all CLI settings accessible from ENV vars has been removed from the project.  Restoring that parity might be a better way to resolve this user's issue.

**Comment by costinm** [comment]
Are only those 3 args that are needed ? We can go case-by-case - either convert those args or use the same pattern we have for agent, check if the files exist ( and define a well-known location - we do support mounting volume as an option in helm ) and just deprecate the 3 CLI args.

Other opinions ? @howardjohn ? I guess we can add the args to helm as well - we have plenty of other options that are rarely used and an even larger number of options and APIs that will not be needed in ambient, maybe we can do an
Istiod args/options cleanup when ambient is ready and not worry for now ?

**Comment by bleggett** [comment]
I agree that ultimately meshconfig or env vars are probably better choices. That being said, as long as istiod supports CLI flags, I think we should let people freely specify them with Helm overrides if they wish to.

**Review by therealmitchconnors — CHANGES_REQUESTED** [review]
After offline conversations with @fsadykov, I will expose these flags as environment variables, which can be set through pilot.env

**Comment by howardjohn** [comment]
https://github.com/istio/istio/pull/42398/files is also adding some args... we may need a consistent approach

**Comment by costinm** [comment]
We have settings in MeshConfig, env and CLI.

I can't imagine a consistent approach where all settings are in the CLI.
Env are a bit easier to work with - but can't support dynamic changes.
MeshConfig allows dynamic changes - and I think is the second best option (
actual CRD / API like we did for telemetry is better if the feature is
broadly used).

I was looking if we can change the code that returns 'env' variables to
peek inside MeshConfig ( using some field like we have in ProxyConfig for
meta / env for sidecar ).
Code seems pretty simple - but the deps are more tricky since the code is
in a different package. Opinions ?
That would make MeshConfig the only source for Istiod.

**Comment by jacob-delgado** [comment]
This has been an issue before for integrating cert-manager and istio-csr (see https://raw.githubusercontent.com/cert-manager/website/master/content/docs/tutorials/istio-csr/example/istio-config-getting-started.yaml).

I'm fine with meshConfig being the source of truth for all of this, but there's been discussions about deprecating that for years now. At this point I'd say it's not really going away. A real CRD would be the best approach, but that would require more resources than Istio has at this point.

Let's see what @therealmitchconnors does as that may be expedient for users, like ourselves, that have to modify charts to allow our users to expose modify these settings to fit their environments.

**Comment by Tenseiga** [comment]
why was this closed. I think the requirement was a valid one.

**Pull request closed without merging**